incubator/hnc/pkg/forest/forest.go

```diff
@@ -142,6 +142,11 @@ func (f *Forest) GetNamespaceNames() []string {
 	return keys
 }
 
+// GetNamespaces returns the namespaces in the forest.
+func (f *Forest) GetNamespaces() namedNamespaces {
+	return f.namespaces
+}
+
```

Comment:
We shouldn't directly return the map like this, since the caller could then modify it directly. This is why `f.namespaces` is unexported (i.e., it has a lowercase first letter) plus the type is unexported too. I'm actually surprised Go let you do this but I guess you never refer to the type `namedNamespaces` directly outside of this package, so it was allowed. 

Instead, the client should use `GetNamespaceNames()`:

```
for _, nsnm := range r.forest.GetNamespaceNames() {
  ns := r.forest.Get(nsnm)
  ...
}

Reply:
Thanks for the explanation. Updated.